File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        
        HashMap<Integer,Integer> map = new HashMap();
        
        Integer n = nums.length;
        
        for( int i=0;i<n;i++ ){
            
            if( map.containsKey(nums[i]) ){
                
                if( i - map.get(nums[i]) <= k ){
                    return true;
                }
                
            }
            
            map.put(nums[i],i);
        }
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        
        map<int,int> m;
        
        int n = (int) nums.size();
        
        for(int i=0; i<n; i++){
            if(m.find(nums[i])!=m.end()){
                if( abs(m[nums[i]]-i)<=k ){
                    return true;
                }
            }
            m[nums[i]] = i;
        }
        return false;
    }
};


